File: aula01/tabelaNomes.lua
indice = 1
tabela = {}

function numeroPar (numero)
  local resultado
  if numero%2 == 0 then
    resultado = true
  else
    resultado = false
  end
  return resultado
end

for linha in io.lines ("nomes.txt") do
  if numeroPar(indice) == false then
    linhaAuxiliar = linha
  else
    tabela[linhaAuxiliar] = linha
  end
  indice = indice + 1
end

for nome, sobrenome in pairs (tabela) do
  print (nome, sobrenome)
end

File: aula01/tabelaNomes.pl
use strict;
use warnings;

my %tabelaNomes;
my $linhaAuxiliar = "";
our $!;

open (my $in, "<", "nomes.txt") or die "Can't open nomes.txt: $!";
my @linhas = <$in>;
my $indice;
sub par {
    my $numero = shift;
    if ($numero%2 == 0) {
        return "true";
    } else {
        return "false";
    }
}

for ($indice=0; $indice< scalar @linhas; $indice++){
    if (par($indice) eq "false"){
        $tabelaNomes {$linhaAuxiliar}=$linhas[$indice];
    } else {
        $linhaAuxiliar = $linhas[$indice];
    }
}

print "$_ $tabelaNomes{$_}\n" for (keys %tabelaNomes);

close $in or die "Can't close nomes.txt: $!";